src/wsproxy.ml

```
@@ -15,124 +15,114 @@
 
 let get_dir_path () = Printf.sprintf "/var/xapi/"
 
-let ignore_exn t () = Lwt.catch t (fun _ -> Lwt.return_unit)
-
 module LwtWsIteratee = Websockets.Wsprotocol(Lwt)
 open Lwt.Infix
 
+
 let start path handler =
   let dir_path = get_dir_path () in
   let fd_sock_path = Printf.sprintf "%s%s" dir_path path in
+  Lwt_log.info_f "Starting wsproxy on %s" fd_sock_path >>= fun () ->
   let fd_sock = Lwt_unix.socket Unix.PF_UNIX Unix.SOCK_STREAM 0 in
   Lwt.catch
     (fun () -> Lwt_unix.unlink fd_sock_path)
-    (fun _ -> Lwt.return ()) >>= fun () ->
+    (fun _ -> Lwt.return_unit) >>= fun () ->
   let () = Lwt_unix.bind fd_sock (Unix.ADDR_UNIX fd_sock_path) in
   let () = Lwt_unix.listen fd_sock 5 in
 
   let rec loop () =
+    let ensure_close = function
+      | [] -> Lwt.return_unit
+      | fds ->
+        Lwt_log.warning_f "Closing %d excess fds" (List.length fds) >>= fun () ->
+        Lwt.catch
+          (fun () -> 
+             List.iter (fun fd -> try Unix.close fd with _ -> ()) fds;
+             Lwt.return_unit)
+          (fun _ -> Lwt.return_unit)
+    in
     Lwt.catch
       (fun () ->
-         Lwt_unix.accept fd_sock >>= fun (fd_sock2,_) ->
          let buffer = String.make 16384 '\000' in
+         Lwt_unix.accept fd_sock >>= fun (fd_sock2,_) ->
          Lwt.catch
```

This is not needed, we already have a catch wrapper and every delicate piece of code is in a finalizer, so we can safely let this fail